subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -2437,6 +2433,43 @@ static bool is_enc_req_pause_tx(struct ll_conn *conn)
 }
 #endif /* CONFIG_BT_CTLR_LE_ENC */
 
+#if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
+/* Connection context pointer used as CPR mutex to serialize connection
+ * parameter requests procedures across simulataneous connections so that
+ * offsets exchanged to the peer does not get changed.
+ */
+static struct ll_conn *conn_upd_curr;
+
+static inline void cpr_mutex_check_and_take(struct ll_conn *conn)
+{
+	if (!conn_upd_curr) {
+		conn_upd_curr = conn;
+	}
+}
+
+static inline void cpr_mutex_take(struct ll_conn *conn)
+{
+	conn_upd_curr = conn;
+}
+
+static inline bool cpr_mutex_is_taken(struct ll_conn *conn)
+{
+	return conn_upd_curr && (conn_upd_curr != conn);
+}
+
+static inline void cpr_mutex_check_and_give(struct ll_conn *conn)
+{
+	if (conn == conn_upd_curr) {
+		conn_upd_curr = NULL;
+	}
+}
+
+static inline void cpr_mutex_give(void)
+{
+	conn_upd_curr = NULL;
+}
```

Comment:
I'm not particularly fond of calling a simple static pointer value for a "mutex". I suggest that you either implement an actual thread-safe mutex, or remove the notation of "mutex" from these functions.

Reply:
This is a connection parameter request procedure mutual exclusion and nothing to do with OS mutex feature. Suggestions on naming are welcome.

@thoh-ot any other alternative to your initial proposal?

Reply:
A simple renaming could be to just do replace `mutex` with `pointer`, and possibly rename `give` to `release`.

Reply:
Renamed to `cpr_active_set` and `cpr_active_reset`